src/main/scala/com/fulcrumgenomics/bam/RandomizeBam.scala

```diff
@@ -58,12 +58,22 @@ class RandomizeBam(
   Io.assertCanWriteFile(output)
 
   override def execute(): Unit = {
-    val extractor  = if (queryGroup) (rec: SamRecord) => rec.name
-                     else            (rec: SamRecord) => rec.name + ":" + rec.basesString + ":" + rec.qualsString
     val in = SamSource(input)
     val hasher = new Murmur3(seed)
-    val keyfunc = (r: SamRecord) => SamOrder.RandomKey(hasher.hashUnencodedChars(extractor.apply(r)), r.asSam.getFlags)
-    val sorter = new Sorter(2e6.toInt, new SamRecordCodec(in.header), keyfunc)
+
+    val sorter = {
+      if (queryGroup) {
+        val f = (r: SamRecord) => SamOrder.RandomQueryKey(hasher.hashUnencodedChars(r.name), r.name, r.asSam.getFlags)
```

Comment:
Why don't you just append the name to the hash?  Then you can go back to the original implementation since `f` has type `SamRecord => SamOrder.RandomKey`?

```suggestion
        val f = (r: SamRecord) => SamOrder. RandomKey(hasher.hashUnencodedChars(r.name) + r.name, r.asSam.getFlags)
```

Reply:
The hash is an `Int` not a `String`.